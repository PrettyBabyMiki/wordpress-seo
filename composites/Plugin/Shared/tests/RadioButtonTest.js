import React from "react";
import renderer from "react-test-renderer";

import IconRadioButton from "../components/RadioButton";
import eye from "../../../../style-guide/svg/eye.svg";
import eyeSlash from "../../../../style-guide/svg/eye-slash.svg";

test( "the unchecked IconRadioButton matches the snapshot", () => {
	const component = renderer.create(
		<IconRadioButton name="group1" id="RadioButton" checkedIcon={ eyeSlash } uncheckedIcon={ eye } checked={ false }/>
	);

	let tree = component.toJSON();
	expect( tree ).toMatchSnapshot();
} );

test( "the checked IconRadioButton matches the snapshot", () => {
	const component = renderer.create(
		<IconRadioButton name="group1" id="RadioButton2" checkedIcon={ eyeSlash } uncheckedIcon={ eye } checked={ true }/>
	);

	let tree = component.toJSON();
	expect( tree ).toMatchSnapshot();
} );
