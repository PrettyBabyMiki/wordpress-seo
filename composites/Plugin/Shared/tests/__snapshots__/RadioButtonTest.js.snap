// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`the checked IconRadioButton matches the snapshot 1`] = `
<span>
  <input
    className="sc-bdVaJa jwdLsH"
    id="RadioButton2"
    name="group1"
    type="radio"
  />
  <label
    checked={true}
    className="label sc-bwzfXH byTXON"
    htmlFor="RadioButton2"
  >
    <svg
      aria-hidden="true"
      className="sc-bxivhb leOCwi"
    />
  </label>
</span>
`;

exports[`the unchecked IconRadioButton matches the snapshot 1`] = `
<span>
  <input
    className="sc-bdVaJa jwdLsH"
    id="RadioButton"
    name="group1"
    type="radio"
  />
  <label
    checked={false}
    className="label sc-bwzfXH gnevpY"
    htmlFor="RadioButton"
  >
    <svg
      aria-hidden="true"
      className="sc-htpNat etIqZE"
    />
  </label>
</span>
`;
